src/draw/shapes.typ

```diff
@@ -582,9 +594,72 @@
   
   // Coordinate check
   pts.map(coordinate.resolve-system)
+
+  // Find the intersection between line a-b next to b
+  // if no intersection could be found, return a.
+  let element-line-intersection(ctx, elem, a, b) = {
+    // Vectors a and b are not transformed yet, but the vectors of the
+    // drawable are.
+    let (ta, tb) = (a, b).map(v => {
+      matrix.mul-vec(ctx.transform, vector.as-vec(v, init: (0,0,0,1)))
+    })
+
+    let pts = ()
+    for drawable in elem.at("drawables", default: ()) {
+      pts += intersection.line-path(ta, tb, drawable)
+    }
+    return if pts == () {
+      a
+    } else {
+      let pt = pts.at(0)
+      let d = vector.dist(pt, b)
+      for i in range(1, pts.len()) {
+        let nd = vector.dist(pts.at(i), b)
+        if nd < d {
+          pt = pts.at(i)
+          d = nd
+        }
+      }
+
+      // Reverse the transformation
+      return matrix.mul-vec(matrix.inverse(ctx.transform),
+        vector.as-vec(pt, init: (0,0,0,1))).slice(0, 3)
+    }
+  }
   
   return (ctx => {
+    let first-elem = pts.first()
+    let first-is-elem = coordinate.resolve-system(first-elem) == "element"
+
+    let last-elem = pts.last()
+    let last-is-elem = coordinate.resolve-system(last-elem) == "element"
```

Comment:
Use the results of `resolve-system` from line 596 above.

Reply:
What do you mean? I am using the result to compare against "element"